### PR TITLE
feat(provider): make client secret non mandatory for public clients

### LINKED ISSUE #153: Modification- Make Client Secret Non Mandatory For Public Clients (Non Blocking Changes)
**Is your feature request related to a problem? Please describe.**
Make Client Secret Non Mandatory For Public Clients (Non Blocking Changes)

**Describe the solution you'd like**
Make Client Secret Non Mandatory For Public Clients (Non Blocking Changes)
**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by Tyagi-Sunny**
## Description

Non blocking changes for making client secret non-mandatory for public clients

Fixes #153 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [x] Performed a self-review of my own code
- [x] npm test passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by yeshamavani** [comment]
PR title 
also add "overrides": {
    "@openapi-contrib/openapi-schema-to-json-schema": "3.2.0"
} in package.json to avoid build errors

**Comment by yeshamavani** [comment]
@Tyagi-Sunny the comments above

**Review by yeshamavani — APPROVED** [review]

File: src/component.ts [hunk 1]
```diff
@@ -47,12 +48,14 @@ import {
...
 } from './strategies/SAML';
-import {AuthenticationConfig} from './types';
+import {AuthenticationConfig, IAuthSecureClientConfig} from './types';
 
 export class AuthenticationComponent implements Component {
   constructor(
     @inject(AuthenticationBindings.CONFIG, {optional: true})
     private readonly config?: AuthenticationConfig,
+    @inject(AuthenticationBindings.SecureClientConfig, {optional: true})
+    private readonly secureClientConfig?: IAuthSecureClientConfig,
```
Comment (samarpan-b) [inline src/component.ts:58]: why new config ? Add it to existing AuthenticationConfig please

**Review by samarpan-b — CHANGES_REQUESTED** [review]

File: src/component.ts (hunk 1 above)
Comment (Tyagi-Sunny) [reply src/component.ts]: changed

**Review by samarpan-b — APPROVED** [review]

**Pull request merged**